backend/src/resolvers.ts

```diff
@@ -0,0 +1,85 @@
+import PostDatasource from "./datasources/post.datasource";
+import UserDatasource from "./datasources/user.datasource";
+import Post from "./entities/post.entity";
+import User from "./entities/user.entity";
+
+const resolvers = {
+    Query: {
+        posts: async (_source, _args, context) => {
+            const postDatasource: PostDatasource = context.dataSources.postDatasource;
+
+            return (await postDatasource.getPosts()).map(post => {
+                return {
+                    ...post, votes:post.voters.length   
+            }});
+        },
+        users: async (_source, _args, context) => {
+            const userDatasource: UserDatasource = context.dataSources.userDatasource;
```

Comment:
```suggestion
            return context.dataSources.userDatasource.getUsers()
```

It's fine to return a promise in a resolver. `apollo-server` takes care of it.